CMakeLists.txt

```diff
@@ -125,10 +125,13 @@ IF(WITH_CORE)
   IF(WITH_SERVER)
     SET (SERVER_SKIP_ECW FALSE CACHE BOOL "Determines whether QGIS server should disable ECW (ECW in server apps requires a special license)")
 
-    SET (WITH_SERVER_PLUGINS TRUE CACHE BOOL "Determines whether QGIS server support for python plugins should be built")
-    IF(WITH_SERVER_PLUGINS)
-      SET(HAVE_SERVER_PYTHON_PLUGINS TRUE)
-    ENDIF(WITH_SERVER_PLUGINS)
+    IF(WITH_BINDINGS)
```

Comment:
Just disabling server plugins when with bindings is off might also be confusing, in the end the problem is that we have two incompatible and conflicting options.

Maybe failing and giving a warning like here would be more appropriate?
https://github.com/qgis/QGIS/blob/700cfe8ab2641864185b09e08c0e2fad31155d8e/CMakeLists.txt#L144

Reply:
Since WITH_SERVER_PLUGINS is ON by default, I would rather set a warning that server plugins are disabled.  
Because plugins support cannot be built without bindings (it is more a matter of hierarchy than conflicting options), I think would prefer that - as a logical dependency - instead requiring to disable plugins explicitely - but that's a matter of philosophy :-)

Reply:
Is the following compatible with both philosophies?

```
IF (WITH_SERVER)
  SET (WITH_SERVER_PLUGINS WITH_BINDINGS CACHE BOOL "Determines whether QGIS server support for python plugins should be built")
  IF(WITH_SERVER_PLUGINS AND NOT WITH_BINDINGS)
    MESSAGE(FATAL_ERROR "Smart message here")
  ENDIF(WITH_SERVER_PLUGINS AND NOT WITH_BINDINGS)
ENDIF (WITH_SERVER)
```

Reply:
I'll go with it, but you still require to set both WITH_SERVER_PLUGINS=FALSE along  with WITH_BINDINGS=FALSE.

Reply:
I don't think you need to state both, `-DWITH_SERVER=FALSE -DWITH_BINDINGS=FALSE` will initialize `WITH_SERVER_PLUGINS` to false, no?

Reply:
-DWITH_SERVER=FALSE disable SERVER_PLUGINS because you disable the whole server support. Furthemore there is no dependency between WITH_BINDINGS and WITH_SERVER.

You still need to state both WITH_SERVER_PLUGINS=FALSE and WITH_BINDINGS=FALSE but anyway its still better than a compile error.

Reply:
> You still need to state both WITH_SERVER_PLUGINS=FALSE and WITH_BINDINGS=FALSE but anyway its still better than a compile error.

I still cannot follow why. In the above example. `WITH_SERVER_PLUGINS` is initialized to the value of `WITH_BINDINGS`, if `WITH_BINDINGS` is false, `WITH_SERVER_PLUGINS` should implicitly also be false without any further ado. Or what am I missing?

Reply:
My bad, I missed 'SET (WITH_SERVER_PLUGINS WITH_BINDINGS CACHE BOOL' !

Reply:
mmh, Infortunately that doesn't seem to work: it seems that WITH_BINDINGS in the 'SET(WITH_SERVER_PLUGINS WITH_BINDINGS...) statement is not resolved as a variable.

Reply:
The right syntax is:
`
SET (WITH_SERVER_PLUGINS ${WITH_BINDINGS} CACHE BOOL "..."
`